test/e2e/testdata/cmp-fileName/plugin.yaml

```diff
@@ -7,4 +7,4 @@ spec:
   generate:
     command: [sh, -c, 'echo "{\"kind\": \"ConfigMap\", \"apiVersion\": \"v1\", \"metadata\": { \"name\": \"$ARGOCD_APP_NAME\", \"namespace\": \"$ARGOCD_APP_NAMESPACE\", \"annotations\": {\"Foo\": \"$FOO\", \"KubeVersion\": \"$KUBE_VERSION\", \"KubeApiVersion\": \"$KUBE_API_VERSIONS\",\"Bar\": \"baz\"}}}"']
   discover:
-    fileName: "subdir/s*.yaml"
+    fileName: "cmp-fileName/subdir/s*.yaml"
```

Comment:
> Can you check why TestCMPDiscoverWithFileName is failing please ? Looks like it is related to changes and not flakiness.

@alexmt It turned out that this test failed because of the test case is wrong.
As stated on the [config management plugin doc](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#write-the-plugin-configuration-file), the `filename` should be applied to the repository root not the application source:
> fileName is a glob pattern (https://pkg.go.dev/path/filepath#Glob) that is applied to the repository's root directory (not the Application source directory). If there is a match, this plugin may be used for the repository.

and in the test case, the `testdata` is the repo, so we need to have `cmp-fileName` as part of the `fileName`.
It was not caught because we only passed `appPath` but not `repoPath` so the `appPath` was considered as `repoPath` and `cmp-fileName` was not needed.

Reply:
This change is causing issues, so I'm going to try to revert this change without reverting your whole bugfix. https://github.com/argoproj/argo-cd/issues/13940